src/core/SampleBuffer.cpp

```diff
@@ -130,6 +130,86 @@ SampleBuffer::SampleBuffer(const f_cnt_t frames)
 
 
 
+SampleBuffer::SampleBuffer(const SampleBuffer& orig)
+{
+	orig.m_varLock.lockForRead();
+
+	m_audioFile = orig.m_audioFile;
+	m_origData = MM_ALLOC(sampleFrame, orig.m_origFrames);
```

Comment:
`malloc` return value for a size 0 allocation is implementation defined, meaning that it could be `nullptr` or a bogus address that can be freed. Do we care that the pointer might not be `nullptr` even though the original is?

It’s a small change anyways if we want them both to be null in that case (just a ternary operator). The same applies to `m_data`.

Reply:
Just a note, that was the last detail I found while reviewing. The rest of the code LGTM. This is actually not even critical: both `SampleBuffer`s will have an empty `m_origData`. The only thing is that one might have a `m_origData == nullptr` and another one might have a `m_origData == 0xDEADBEEF` or whatever value is used to represent that empty buffer.

I still think it's worth fixing because it's simple to fix it and that could cause hard to track bugs if we compare that pointer with `nullptr` or compare the members two `SampleBuffer`s to check if they are equal. Something like `m_origData = orig.m_origData ? MM_ALOC(sampleFrame, orig.m_origFrames) : nullptr;` should do it.

Reply:
Ought to be fixed in latest commit.